components-rs/bytes.rs

```diff
@@ -0,0 +1,301 @@
+use datadog_trace_utils::span::SpanBytes;
+use ddcommon_ffi::slice::{AsBytes, CharSlice};
+use std::borrow::Cow;
+use tinybytes::{Bytes, BytesString, UnderlyingBytes};
+
+/// cbindgen:no-export
+
+#[repr(C)]
+pub struct ZendString {
+    pub refcount: u32,
+    pub type_info: u32,
+    pub h: u64,
+    pub len: usize,
+    pub val: [u8; 1],
+}
+
+struct ZendStringWrapper(*mut ZendString);
+
+unsafe impl Send for ZendStringWrapper {}
+unsafe impl Sync for ZendStringWrapper {}
+
+impl AsRef<[u8]> for ZendStringWrapper {
+    fn as_ref(&self) -> &[u8] {
+        unsafe { (*self.0).as_ref() }
+    }
+}
+
+impl AsRef<[u8]> for ZendString {
+    fn as_ref(&self) -> &[u8] {
+        unsafe { std::slice::from_raw_parts((*self).val.as_ptr(), (*self).len) }
+    }
+}
+
+impl Drop for ZendStringWrapper {
+    fn drop(&mut self) {
+        extern "C" {
+            fn ddog_zend_string_release(s: *mut ZendString);
+        }
+
+        unsafe {
+            ddog_zend_string_release(self.0);
+        }
+    }
+}
+
+impl UnderlyingBytes for ZendStringWrapper {}
+
+unsafe fn convert_to_bytes(zend_str: *mut ZendString) -> Bytes {
+    (*zend_str).refcount += 1; // Increment the reference count to prevent double free
+
+    match String::from_utf8_lossy((*zend_str).as_ref()) {
+        Cow::Owned(s) => s.into(),
+        Cow::Borrowed(_) => Bytes::from_underlying(ZendStringWrapper(zend_str)),
+    }
```

Comment:
```suggestion
    Bytes::from_underlying(ZendStringWrapper(zend_str)),
```
We only need from_utf8_lossy if we actually need a BytesString. For raw Bytes that's not needed.